src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -85,7 +85,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 		for( int x = 0; x < inTag.tagCount(); x++ )
 		{
-			final ItemStack gs = new ItemStack( inTag.getCompoundTagAt( x ) );
+			NBTTagCompound ingredient = inTag.getCompoundTagAt( x );
+			final ItemStack gs = new ItemStack( ingredient );
+
+			if( !ingredient.hasNoTags() && gs.isEmpty() )
+			{
+				throw new IllegalArgumentException( "No pattern here!" );
```

Comment:
I am still unsure about the fast fail approach here. First it validates `ingredient` after it is actually used to create the itemstack. Thus it is actually done too later and should also be redundant as the itemstack should also be empty.

Thus should it actually be needed (and why?), it should probably just fail on `gs.isEmpty()` as this already includes the `hasNoTags()`, but this itself is not sufficient to detect an invalid itemstack before creating it.